src/Microsoft.AspNetCore.Server.Kestrel/Http/MessageBody.cs

```diff
@@ -353,12 +462,34 @@ private static bool TakeChunkedLine(SocketInput baton, ref int chunkSizeOut)
                 }
             }
 
+            private static void ThrowInvalidFormat()
+            {
+                throw new InvalidOperationException("Bad Request");
+            }
+
+            private static void ThrowChunkedRequestIncomplete()
+            {
+                throw new InvalidOperationException("Chunked request incomplete");
```

Comment:
I'm not sure. `FormatException` seems more appropriate in places where you're using format strings, or parsing a string to convert it to some other type.

In our case I think we should use:

* `InvalidOperationException` where things are just wrong (like incomplete request).
* `NotSupportedException` when something is not supported (like chunked extensions).